Add react-toastify to change notification logic.

tuovinensanttu opened the pull request:
## Description
<!-- Describe your changes in detail -->
Installed `react-toastify` and wrapper the old `NotificationComponent` with it. 

## Context
<!-- Why is this change required? What problem does it solve? -->
<!-- Leave a link to the Jira ticket for posterity. -->
This change allows us to get rid of the logic where `NotificationComponent` was imported to almost every component.


## How Has This Been Tested?
Tested this manually.

## Manual Testing Instructions for Reviewers
<!-- Make it easy for reviewers to test your changes by providing instructions -->
One way to test this is to use renew membership functionality.

tuovinensanttu commented on src/domain/auth/components/login/Login.tsx:
```diff
@@ -42,6 +42,10 @@ function Login(props: Props) {
 
   const isAuthenticated = useSelector(isAuthenticatedSelector);
 
+  if (Boolean(props.auth.error)) {
+    toastNotification({ onClose: () => props.resetError() });
+  }
```
This solution seems to be working, only one notification is shown at the time. I'm still thinking if I should wrap this with `useEffect` to be sure.

Jokinen replied:
Well, you are correct. The render function should not have side effects unless wrapped in a `useEffect`. But I think that an even better route could be to set the toast in the authentication routine itself, in which case you would not need to use a `useEffect` either. Now the firing of the notification doesn't need to be tied to this component anymore.

Jokinen commented on CHANGELOG.md:
```diff
@@ -36,6 +36,7 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 ### Changed
 
 - Use GitHub actions instead of Travis
+- centralize notification with react-toastify.
```
Minor, no capitalisation

tuovinensanttu replied:
Fixed!

Jokinen commented on src/common/components/notification/toastNotification.tsx:
```diff
@@ -0,0 +1,23 @@
+  onClose?: () => void;
+};
+
+const toastNotification = (notificationOptions: NotificationOptions) => {
```
This is stored under `components`. Is it really a component?

tuovinensanttu replied:
True! Moved it under helpers instead.

Jokinen commented on src/domain/app/App.tsx:
```diff
@@ -19,7 +21,7 @@ import authConstants from '../../domain/auth/constants/authConstants';
 import AppMeta from './AppMeta';
 import AppRoutes from './AppRoutes';
-
+import styles from './app.module.css';
```
Missing whitespace.

Jokinen commented on src/domain/app/AppYouthProfileRoute.tsx:
```diff
@@ -29,12 +29,9 @@ function AppYouthProfileRoute({
-    onError: () => {
-      setShowNotification(true);
-    },
+    onError: () => toastNotification({}),
```
Using an empty object is discouraged.

tuovinensanttu replied:
Fixed now. Silly mistake on my part to, forgot to make whole `notificationOptions` optional.

Jokinen reviewed (commented):
Good changes! Added a few comments.

tuovinensanttu commented:
> Well, you are correct. The render function should not have side effects unless wrapped in a useEffect. But I think that an even better route could be to set the toast in the authentication routine itself, in which case you would not need to use a useEffect either. Now the firing of the notification doesn't need to be tied to this component anymore.

Seems that I was so used to notification being a component that I didn't even think about this. Removed error handling from `Login` component and moved it to `authenticate` and `logout` methods.
